Kanban4U/GitHub/GitHubLogic.cs

```diff
@@ -0,0 +1,306 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web;
+using Windows.Security.Authentication.Web;
+using Windows.UI.Xaml.Controls;
+using Windows.Web.Http;
+using Windows.Storage.Streams;
+using Windows.Web.Http.Headers;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Threading;
+using Windows.UI;
+using Windows.UI.Xaml.Markup;
+
+namespace Kanban4U
+{
+    public static class GitHubLogic
+    {
+        public static async Task<string> GetGitHubAccessToken()
+        {
+            var settings = GlobalSettings.Instance;
+            if (settings.GitHubAccessToken == null)
+            {
+                settings.GitHubAccessToken = await GetGitHubAccessTokenWorker();
```

Comment:
Having the side effect of this makes this not feel like an honest "Get" method. "UpdateGitHubAccessToken" maybe?

Reply:
Changed to "Ensure"